includes/onnxruntime_customop.hpp

```diff
@@ -100,18 +62,44 @@ template <typename TOp, typename TKernel>
 struct CustomOpBase : OrtCustomOp {
   CustomOpBase() {
     OrtCustomOp::version = 10;  // The minimum ORT version supported
-    OrtCustomOp::CreateKernel = [](const OrtCustomOp* this_, const OrtApi* api, const OrtKernelInfo* info) { return static_cast<const TOp*>(this_)->CreateKernel(*api, info); };
-    OrtCustomOp::GetName = [](const OrtCustomOp* this_) { return static_cast<const TOp*>(this_)->GetName(); };
-
-    OrtCustomOp::GetExecutionProviderType = [](const OrtCustomOp* this_) { return static_cast<const TOp*>(this_)->GetExecutionProviderType(); };
+    OrtCustomOp::CreateKernel = [](const OrtCustomOp* this_, const OrtApi* api, const OrtKernelInfo* info) {
+      void* result = nullptr;
+      API_IMPL_BEGIN
+      result = static_cast<const TOp*>(this_)->CreateKernel(*api, *info);
+      API_IMPL_END
+      return result;
```

Comment:
Probably. I was contemplating adding new CreateKernel/Compute functions to the ORT CustomOp to allow returning a status when I set it up this way, but as we're just re-throwing or calling abort() it may be fine (if the compiler knows an `abort()` means no return is required).

Reply:
Yeah nah. It's not smart enough to treat the 'abort()' as a return path so generates a warning (at least with MSVC)